lib/node.js

```diff
@@ -299,6 +310,90 @@ class Node {
     this.edgesIn.forEach(edge => edge.reload(true))
   }
 
+  // node.explain(nodes seen already, edge we're trying to satisfy
+  // if edge is not specified, it lists every edge into the node.
+  explain (edge = null, seen = []) {
+    if (this[_explanation])
+      return this[_explanation]
+
+    return this[_explanation] = this[_explain](edge, seen)
+  }
+
+  [_explain] (edge, seen) {
+    if (this.isRoot && !this.sourceReference) {
+      return {
+        location: this.path
+      }
+    }
+
+    const why = {
+      name: this.isRoot ? this.package.name : this.name,
```

Comment:
I think we should make a getter for this and never have to care about it again 😂 :
```
get realname() {
  return this.isRoot ? this.package.name : this.name
}
```